cpp/src/cider/exec/nextgen/function/CiderStringFunctions.cpp

```diff
@@ -177,6 +207,25 @@ extern "C" ALWAYS_INLINE int64_t cider_concat(char* string_heap_ptr,
   return pack_string_t(s);
 }
 
+extern "C" ALWAYS_INLINE int32_t cider_concat_len(int lhs_len, int rhs_len) {
```

Comment:
What's the purpose for functions `cider_concat_len`, `cider_ascii_upper_len` and `cider_ascii_lower_ptr`? seems they are not necessary.

Reply:
yes, it's not necessary, we can calculate the result size directly in xxxStringExpr::codegen method. keep these functions may help understand. By the way, this functions can be optimized.